src/Microsoft.ML.Transforms/Text/TextStaticExtensions.cs

```diff
@@ -113,4 +115,579 @@ public bool Equals(Reconciler other)
+                    _invertHash == other._invertHash;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var pairs = new List<(string[] inputs, string output)>();
+                foreach (var outCol in toOutput)
+                    pairs.Add((new[] { inputNames[((OutPipelineColumn)outCol).Input] }, outputNames[outCol]));
+
+                return new NgramHashEstimator(env, pairs.ToArray(), _hashBits, _ngramLength, _skipLength, _allLengths, _seed, _ordered, _invertHash);
+            }
+        }
+
+        /// <summary>
+        /// Produces a bag of counts of ngrams (sequences of consecutive words of length 1-n) in a given tokenized text.
+        /// It does so by hashing each ngram and using the hash value as the index in the bag.
+        ///
+        /// <see cref="ToNgramsHash"/> is different from <see cref="WordHashBagEstimatorExtensions.ToBagofHashedWords"/>
+        /// in a way that <see cref="ToNgramsHash"/> takes tokenized text as input while <see cref="WordHashBagEstimatorExtensions.ToBagofHashedWords"/> tokenizes text internally.
+        /// </summary>
+        /// <param name="input">The column to apply to.</param>
+        /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
+        /// <param name="ngramLength">Ngram length.</param>
+        /// <param name="skipLength">Maximum number of tokens to skip when constructing an ngram.</param>
+        /// <param name="allLengths">Whether to include all ngram lengths up to <paramref name="ngramLength"/> or only <paramref name="ngramLength"/>.</param>
+        /// <param name="seed">Hashing seed.</param>
+        /// <param name="ordered">Whether the position of each source column should be included in the hash (when there are multiple source columns).</param>
+        /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
+        public static Vector<float> ToNgramsHash(this VarVector<Key<uint, string>> input,
+            int hashBits = 16,
+            int ngramLength = 2,
+            int skipLength = 0,
+            bool allLengths = true,
+            uint seed = 314489979,
+            bool ordered = true,
+            int invertHash = 0) => new OutPipelineColumn(input, hashBits, ngramLength, skipLength, allLengths, seed, ordered, invertHash);
+    }
+
+    /// <summary>
+    /// Extensions for statically typed <see cref="LdaEstimator"/>.
+    /// </summary>
+    public static class LdaEstimatorExtensions
```

Comment:
>LdaEstimatorExtensions [](start = 24, length = 22)

Please only review LDA related classes. The other text transform changes are being reviewed in #953. I will merge the branches once approved.

Reply:
merged.


---
In reply to: [219356989](https://github.com/dotnet/machinelearning/pull/972#discussion_r219356989) [](ancestors = 219356989)